projects/observability/src/shared/dashboard/widgets/charts/cartesian-widget/cartesian-widget.model.ts

```diff
@@ -24,7 +26,7 @@ import { MetricSeries, MetricSeriesDataFetcher, SeriesModel } from './series.mod
   type: 'cartesian-widget',
   displayName: 'Cartesian Widget'
 })
-export class CartesianWidgetModel<TData> {
+export class CartesianWidgetModel {
```

Comment:
Note the generic removed here. This is needed due to the datasource only providing an array of intervals. This means the baseline and series data gets combined in the data source model, then pulled apart again here in the CartesianWidgetModel so it can be separated into different series and charted correctly.

Not really in love with what I have here, so open to other suggestions.

Reply:
It feels to me like some of these hoops are based on the new json model that we landed on here. Do you mind sharing an example of a full widget with the new model for comparison?

I was thinking it would end up something like below. I'm not positive this is any better since I haven't been working in the code enough to see any pitfalls/how much work this change would be/how much it would clean up, but since you were open to suggestions I figured I'd throw it out there. For future changes like this, I often find that starting with the desired json model is a good starting point - similar to proposing an api before implementing it.

```typescript
            {
                type: 'cartesian-widget',
                ...
                series: [
                  {  // Regular data
                    type: 'series',
                    ...
                    data: {
                      type: 'entity-metric-timeseries-data-source',
                      metric: {
                        type: 'metric-timeseries',
                        metric: 'duration',
                        aggregation: MetricAggregationType.P99
                      }
                    }
                  },
                  { // Baseline data
                    type: 'series',
                    'visualization-type': 'dashed',
                    ...
                    data: {
                      type: 'entity-metric-timeseries-data-source',
                      metric: {
                        type: 'metric-baseline-timeseries', // note the different spec
                        metric: 'duration',
                        aggregation: MetricAggregationType.P99
                      }
                    }
                  }
              ],
              bands: [
                { // Band data
                    type: 'band',
                    ...
                    data: {
                      type: 'entity-metric-band-timeseries-data-source',
                      metric: {
                        type: 'metric-baseline-timeseries',
                        metric: 'duration',
                        aggregation: MetricAggregationType.P99
                      }
                    }
                  }
```

Reply:
```typescript
export const serviceMetricsExtendedDashboard: DashboardDefaultConfiguration = {
  ...serviceMetricsDashboard,
  json: {
    type: 'container-widget',
    layout: {
      type: 'auto-container-layout',
      rows: 4,
      'enable-style': false
    },
    children: [
      {
        type: 'cartesian-widget',
        'selectable-interval': true,
        'legend-position': LegendPosition.None,
        series: [
          {
            type: 'series',
            name: 'Average Duration',
            color: '#0076C7',
            'visualization-type': SeriesVisualizationType.Line,
            band: {
              type: 'band',
              name: ''
            },
            data: {
              type: 'entity-metric-timeseries-data-source',
              metric: {
                type: 'metric-timeseries',
                metric: 'duration',
                aggregation: MetricAggregationType.Average,
                showBands: true
              }
            }
          }
        ]
      }
    ]
  }
};
```

Reply:
Yeah, I started with what you have there and changed it to deal with that bug that I didn't know the timeline on. Now that the bug is fixed, I'm not too happy with how this workaround way turned out.

Let me see how much work it is to do this the right way.